### PR TITLE
Migrate basic entity persister to the object API of association mapping

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/ClassMetadata.php [hunk 1]
```diff
@@ -438,7 +438,7 @@ class ClassMetadata implements PersistenceClassMetadata, Stringable
      * )
      * </pre>
      *
-     * @psalm-var array<string, OneToOneAssociationMapping|ManyToOneAssociationMapping|OneToManyAssociationMapping|ManyToManyAssociationMapping>
+     * @psalm-var array<string, AssociationMapping>
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Mapping/ClassMetadata.php:441]: I had to revert to something more simple here. It confused Psalm too much, especially when performing intersections with owning types.

**Comment by greg0ire** [comment]
ping @derrabus @SenseException :pray:

**Review by SenseException — APPROVED** [review]
Sorry for forgetting about this PR. I started a review back then late at night and wanted to check `BasicEntityPersister` on the next day but forgot to mark the PR as unread. :pray:

**Comment by greg0ire** [comment]
No worries :)

**Pull request merged**